pyqg/kernel.pyx

```diff
@@ -72,9 +72,9 @@ cdef class PseudoSpectralKernel:
     cdef readonly DTYPE_com_t [:] _il
     cdef public DTYPE_real_t [:,:] _k2l2
     # background state constants (functions of z only)
-    cdef DTYPE_real_t [:] Ubg
-    cdef DTYPE_real_t [:] Qy
-    cdef readonly DTYPE_com_t [:, :] _ikQy
+    cdef DTYPE_real_t [:,:] Ubg
+    cdef DTYPE_real_t [:,:] Qy
+    cdef readonly DTYPE_com_t [:, :, :] _ikQy
```

Comment:
Just trying to understand your approach.
It looks like you have added another dimension to the background state, which is exactly what is needed in order to implement this feature correctly.